runtime/libia2/include/ia2.h

```diff
@@ -48,7 +48,9 @@
 #define IA2_CALL(opaque, id) opaque
 #define IA2_CAST(func, ty) (ty) (void *) func
 #else
-#define IA2_DEFINE_WRAPPER(func) IA2_DEFINE_WRAPPER_##func
+#define IA2_DEFINE_WRAPPER(func) \
+    __attribute__((__used__)) static void *keep_##func = (void *)func; \
```

Comment:
I don't love adding a function pointer to data just to mark a function as used. We don't have any other way?

Reply:
I haven't put much thought into this. There might be a way to get `__attribute__((used)) typeof(func) func;` before the function definition, but I'd need to think about how that would work with the types `func`'s signature depends on.